packages/ui/src/components/Visualization/Canvas/canvas.service.ts

```diff
@@ -76,9 +105,21 @@ export class CanvasService {
       case LayoutType.Concentric:
         return new ConcentricLayout(graph);
       case LayoutType.DagreVertical:
-        return new DagreLayout(graph, { rankdir: 'TB', nodesep: 20, ranksep: 0 });
+        return new DagreLayout(graph, {
+          rankdir: 'TB',
+          ranker: 'network-simplex',
+          nodesep: 20,
+          edgesep: 20,
+          ranksep: 20,
```

Comment:
Probably this needs to be `0` to keep the edges short
```suggestion
          ranksep: 0,
```